### PR TITLE
Update interface to satisfy torch.compile being goofy.

### PR DESCRIPTION
**Pull request by CoffeeVampir3**
There's a strange edge case from torch compile that seems to not be happy with `s != None:` but is perfectly content with `s is not None:` so I've made the change to satisfy torch compile here. Simple alteration. 

Here's the error for reference:

```cpp
  File "/home/blackroot/.pyenv/versions/3.12.7/lib/python3.12/site-packages/torch/_dynamo/variables/builder.py", line 2333, in wrap_fx_proxy_cls
    unimplemented(
  File "/home/blackroot/.pyenv/versions/3.12.7/lib/python3.12/site-packages/torch/_dynamo/exc.py", line 297, in unimplemented
    raise Unsupported(msg, case_name=case_name)
torch._dynamo.exc.Unsupported: torch.* op returned non-Tensor bool call_function <built-in function ne>

from user code:
   File "/home/blackroot/Desktop/minimodels/lm lightning/models/model.py", line 44, in forward
    x = block(x)
  File "/home/blackroot/Desktop/minimodels/lm lightning/models/lightning_tensor_product.py", line 227, in forward
    x = self.learned_residual_scale_attn * x + self.attn(x)
  File "/home/blackroot/Desktop/minimodels/lm lightning/models/lightning_tensor_product.py", line 173, in forward
    attn_out = lightning_attn_func(
  File "/home/blackroot/.pyenv/versions/3.12.7/lib/python3.12/site-packages/lightning_attn/ops/lightning_attn_interface.py", line 50, in lightning_attn_func
    if s != None:
```

This might be an edge case inductor bug but I'll commit this fix on the offchance this is intended behaviour.

Reference torch compile settings:
```
    model = torch.compile(
        model,
        backend='inductor',
        dynamic=False,
        fullgraph=True,
        options={
            "epilogue_fusion": True,
            "max_autotune": True,
        }
    )
 ```

### CODE REVIEW

**Comment by Doraemonzzz** [comment]
Thank you for the pull requests you provided. Could you please provide a minimal script to reproduce the issue? It would help me with some testing.

**Comment by CoffeeVampir3** [comment]
Here's a minimal test:

```python
import torch
from torch import nn

class LightningAttention(nn.Module):
    def forward(self, query, key=None, value=None, mask=None):
        batch_size, seq_len, dim = query.shape
        
        scores = torch.matmul(query, key.transpose(-2, -1)) / (dim ** 0.5)
        if mask != None:
            scores = scores + mask
            
        attn = torch.softmax(scores, dim=-1)
        return torch.matmul(attn, value)

batch_size, seq_len, dim = 2, 4, 8
x = torch.randn(batch_size, seq_len, dim)

model = LightningAttention()
try:
    model = torch.compile(
        model,
        backend='inductor',
        dynamic=False,
        fullgraph=True,
        options={
            "epilogue_fusion": True,
            "max_autotune": True,
        }
    )
    mask = torch.zeros(batch_size, seq_len, seq_len)
    output = model(x, x, x, mask)
except Exception as e:
    print(f"Error: {e}")
```

It seems very specific to my compile settings, I've also opened a bug report with torch [issue](https://github.com/pytorch/pytorch/issues/145277)

**Comment by Doraemonzzz** [comment]
I will reproduce this issue today, but it seems that softmax attention is being used here. Should I change it to lightning attention?

**Comment by CoffeeVampir3** [comment]
It looks like this is a confirmed problem on torch's end, I'll close this :+1:
